### PR TITLE
melt Ne

### PR DESCRIPTION
**Pull request by tranma**
just did this https://github.com/ambiata/icicle/issues/214

also updated CLI tests, probably was made out of date from the constant folding branch

### CODE REVIEW

File: src/Icicle/Avalanche/Statement/Simp/Constructor.hs [hunk 1]
```diff
@@ -211,17 +217,24 @@ constructor a_fresh statements
...
 
    | otherwise
    = x
 
+  relation Min.PrimRelationGt = primGt
+  relation Min.PrimRelationGe = primGe
+  relation Min.PrimRelationLt = primLt
+  relation Min.PrimRelationLe = primLe
+  relation Min.PrimRelationEq = primEq
+  relation Min.PrimRelationNe = primNe
```
Comment (amosr) [inline src/Icicle/Avalanche/Statement/Simp/Constructor.hs]: Is this right?
```
(a,b) /= (c,d)
```
should really be
```
a /= c || b /= d 
```

and the less than is more complicated, I think

Comment (tranma) [inline src/Icicle/Avalanche/Statement/Simp/Constructor.hs]: oh yeah, should have actually thought about it for a second

Comment (amosr) [inline src/Icicle/Avalanche/Statement/Simp/Constructor.hs]: it's more annoying than it looks! less than is something like..

```
(a,b) < (c,d)
==>
(a < c) || (a == c && b < d)
```
I think.
I don't think ord is particularly important but if /= is easy to fix might as well do that

**Comment by tranma** [comment]
I'll just merge this? Ord can come later after psv stuff is done

**Comment by amosr** [comment]
yup, sounds good :100:

**Pull request merged**